docs/notebooks/asynchronous_bayesian_optimization.pct.py

```diff
@@ -0,0 +1,373 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.12.0
+#   kernelspec:
+#     display_name: 'Python 3.7.5 64-bit (''.venv'': venv)'
+#     name: python3
+# ---
+
+# %% [markdown]
+# # Asynchronous Bayesian optimization with Trieste
+#
+# In this notebook we show how Bayesian optimization can be done asynchronuosly. That is pertinent in scenarios when the objective function we are aiming to optimize can be run for several points in parallel, and observations might return back at different times. To avoid wasting resources we immediatelly request the next point asynchronuosly, taking into account points that still being evaluated.
+#
+# To contrast this approach with regular [batch optimization](batch_optimization.ipynb), this notebook also shows how to run parallel synchronous batch approach.
+
+# %%
+# silence TF warnings and info messages, only print errors
+# https://stackoverflow.com/questions/35911252/disable-tensorflow-debugging-information
+import os
+
+os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
+import tensorflow as tf
+
+tf.get_logger().setLevel("ERROR")
+import numpy as np
+import math
+
+import time
+import timeit
+
+
+# %% [markdown]
+# First, let's define a simple objective that will emulate workload of variable time. We will be using class BO benchmark function [Branin](https://www.sfu.ca/~ssurjano/branin.html), and insert sleep call in the middle of the calculation to emulate delay. Our sleep delay is a scaled sum of all input values to make sure delays are uneven.
+# %%
+def objective(points, sleep=True):
+    if points.shape[1] != 2:
+        raise ValueError(f"Incorrect input shape, expected (*, 2), got {points.shape}")
```

Comment:
Minor: following TensorFlow syntax for tensor shapes.
```suggestion
        raise ValueError(f"Incorrect input shape, expected [None, 2], got {points.shape}")
```